File: ASP.NETCoreWebApplicationWithAngular/ClientApp/src/app/shared/BucketBook.ts
import {User} from "./User";

export class BucketBook {
  constructor(

    public bucketBookId?: number,
    public name?: string,
    public author?: string,
    public description?: string,
    public price?: number,
    public count?: number,
    public userId?: number,
    public user?: User
  ) {
  }
}

File: ASP.NETCoreWebApplicationWithAngular/Models/BucketBook.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace ASP.NETCoreWebApplicationWithAngular.Models
{
    public class BucketBook
    {
        public int BucketBookId { get; set; }
        public string Name { get; set; }
        public string Author { get; set; }
        public string Description { get; set; }
        public int Count { get; set; }
        public int Price { get; set; }
        public int UserId { get; set; } 
        public User User { get; set; }
    }
}